wiki/Introduction-to-Creating-a-Job.md

```diff
@@ -0,0 +1,1818 @@
+installed. This CLI provides easy to use subcommands for validating the syntax of a Job Template, running Tasks defined by a Job
+Template locally on your workstation, and more. The CLI is written in Python, so you will need to have Python 3.9 or higher
+available on your workstation; for information on how to install Python, please see the official [Python.org website](https://www.python.org/).
+
+First, we suggest creating a [Python virtual environment](https://docs.python.org/3/library/venv.html) and installing the CLI into that:
+
+| POSIX | Windows (CMD Shell) |
+| ----- | ------- | 
+| <pre>python3 -m venv .venv<br>source .venv/bin/activate<br>pip install openjd-cli</pre> | <pre>python3 -m venv .venv<br>.venv\Scripts\activate.bat<br>pip install openjd-cli</pre> | 
+
+For writing Job Templates by hand we also recommend using [Visual Studio Code](https://code.visualstudio.com/)
+and configuring it so that it can auto-complete the syntax of your Job Templates. To configure auto-complete, you
+will first need to generate schema files for Open Job Description using the CLI:
+
+```bash
+mkdir ~/openjd-schemas
+cd ~/openjd-schemas
+openjd schema --version jobtemplate-2023-09 > openjobdescription-jobtemplate-2023-09.json
+openjd schema --version environment-2023-09 > openjobdescription-environment-2023-09.json
+```
+
+Then install the official [Red Hat YAML](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-yaml) extension, and 
+modify your [Visual Studio Code settings](https://code.visualstudio.com/docs/languages/json#_json-schemas-and-settings)
+to apply the Open Job Description schemas to your Job and Environment Templates. The following is an example of settings to
+automatically apply the schemas to all JSON and YAML files on a workstation:
+
+```json
+    "json.schemas": [
+        {"fileMatch": ["/**/*.json"], "url": "/Users/<username>/openjd-schemas/openjobdescription-jobtemplate-2023-09.json"},
+        {"fileMatch": ["/**/*.json"], "url": "/Users/<username>/openjd-schemas/openjobdescription-environment-2023-09.json"}           
+    ],
+    "yaml.schemas": {
+        "/Users/<username>/json-schemas/openjobdescription-jobtemplate-2023-09.json": [
+            "/**/*.yaml"
+        ],
+        "/Users/myusername/json-schemas/openjobdescription-environment-2023-09.json": [
+            "/**/*.yaml"
+        ]
+    }
+```
+
+To use auto-complete, simply use Control-Space and select a schema entity to automatically populate all of the entities required
+fields, or select a field to auto-complete it:
+
+| Menu | Selecting "JobTemplate" Entity | Selecting "specificationVersion" field |
+| ---  | --- | ---- |
+| ![Autocomplete menu](images/template_autocomplete_menu.png) | ![Autocomplete template](images/template_autocomplete_entity.png) | ![Autocomplete field](images/template_autocomplete_field.png) |
```

Comment:
Why are we still getting red underline errors with the specification installed?

Reply:
Mostly that the values are blank, so vs-code is highlighting that. The highlight on the standalone specificationVersion line is because there are other required fields that aren't yet present.